### PR TITLE
feature/adding_mermaid_renderer_to_markdownMacro

### PR DESCRIPTION
**Pull request by ateixeirafr**

### CODE REVIEW

**Comment by boris-moduscreate** [comment]
@khanguyen88 / @nhan-nguyen-se not sure which of you two has more experience in this area. How does this align with our Cloud implementation? Will this be migrate-able to Cloud in a sane way?

File: src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java [hunk 1]
```diff
@@ -228,6 +229,20 @@ public String execute(Map<String, String> parameters, String bodyContent, Conver
...
+                "var config = {\n"+
+                "    startOnLoad:true,\n"+
+                "    theme: 'default',\n"+
+                "    flowchart:{\n"+
+                "            useMaxWidth:false,\n"+
+                "            htmlLabels:true\n"+
+                "        }\n"+
+                "};\n"+
+                "mermaid.initialize(config);\n"+
+                "window.mermaid.init(undefined, document.querySelectorAll('.language-mermaid'));\n"+
```
Comment (khanguyen88) [inline src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java:243]: ``.language-mermaid`` could to be to broad and might include elements from other macros. Let's use something like this to limit the scope to just MardownMacro:
```
[data-macro-name=\"markdown-from-url\"] .language-mermaid'
```

File: src/main/resources/atlassian-plugin.xml [hunk 2]
```diff
@@ -89,6 +89,11 @@
         <resource type="download" name="highlight.min.css" location="css/highlight.min.css" />
     </web-resource>
     
+    <web-resource key="mermaidjs" name="mermaid.js" >
+        <resource type="download" name="mermaid.min.js" location="js/mermaid.min.js" />
```
Comment (khanguyen88) [inline src/main/resources/atlassian-plugin.xml:93]: [nitpick]: these files should have the version in file name, so we know which version we're using, eg: mermaid-1.0.1.min.js

**Comment by khanguyen88** [comment]
> @khanguyen88 / @nhan-nguyen-se not sure which of you two has more experience in this area. How does this align with our Cloud implementation? Will this be migrate-able to Cloud in a sane way?

This PR only changes how the macro is rendered, not how it is stored, so should be compatible with existing Cloud Migration logic.

**Comment by boris-moduscreate** [comment]
> This PR only changes how the macro is rendered, not how it is stored, so should be compatible with existing Cloud Migration logic.

What is the effort going to be to support migrating this on the Cloud side? I expect we would need that done before we could release this.

**Comment by khanguyen88** [comment]
@boris-moduscreate I did a some testing and the macro is migrated fine, no code change needed (found a bug in our migration logic, but it's unrelated to this PR)

@ateixeirafr during my testing, the diagram you included in the sample file did not render properly (invalid syntax), but it rendered fine on mermaidjs site. So maybe a bug here

**Comment by boris-moduscreate** [comment]
@khanguyen88 Can you file a relevant bug and I can prioritize?

@ateixeirafr let us know once you have fixed the issue Kha found.

**Comment by ateixeirafr** [comment]
Hello Friends, 

The issue in example file is corrected.

Regards.

**Comment by boris-moduscreate** [comment]
@khanguyen88 does this look right? or was there actually an error on the render side? Asking since you mentioned that the old mermaid code worked in another renderer?

**Comment by khanguyen88** [comment]
@boris-moduscreate 
The old sample worked on mermaidjs site https://mermaid.live/edit.
The latest version works fine on both Server and Cloud version. I tested on my local Server and migrated to devfirst here https://devfirst.atlassian.net/wiki/spaces/KHA/pages/240943115/Markdown+Local

File: src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java (hunk 1 above)
Comment (khanguyen88) [inline src/main/java/com/atlassian/plugins/confluence/markdown/MarkdownFromURLMacro.java:243]: .language-mermaid could to be too broad and might include elements from other macros. Let's use something like this to limit the scope to just MardownMacro:
```
[data-macro-name=\"markdown-from-url\"] .language-mermaid'
```
This should be fixed on both macros

**Comment by ateixeirafr** [comment]
Hello, @khanguyen88  
Please check what I done for limiting the scope.

The "Syntax error in graph" issue i identified is due to the hexagon shape rendering
I don't know why but an empty xml comment node is injected between the two opening "curly braces"
for example 
- mh{{myHexagon Shape}} 
will render as
- mh{{myHexagon Shape}}

Regards

**Review by khanguyen88 — APPROVED** [review]

**Comment by khanguyen88** [comment]
LGTM, thank you for your PR 👍

**Comment by boris-moduscreate** [comment]
1.6.19 has been released with these changes.

**Pull request merged**